api/include/opentelemetry/common/spin_lock_mutex.h

```diff
@@ -101,9 +101,7 @@ class SpinLockMutex
         __builtin_ia32_pause();
 #  endif
 #elif defined(__arm__)
-        // This intrinsic should fail to be found if YIELD is not supported on the current
-        // processor.
-        __yield();
```

Comment:
Is there a way (e.g. macro or something) to determine if `__yield` is available?

Reply:
Curious where this pattern comes from.

BTW, do we need add CI for arm?

Reply:
> Is there a way (e.g. macro or something) to determine if __yield is available?

There is no standard approach to do this. ARM vendors may choose not to implement these extensions.

> Curious where this pattern comes from.

This patten of yielding ? It was written by Josh. I would prefer.to let these platform specific code go away as long term solution, and rely on yield provided by standard C++ library.

> BTW, do we need add CI for arm?

Yes we need it.  @esigo has done some initial work https://github.com/open-telemetry/opentelemetry-cpp/pull/1165, but seems builds are very slow, as there is no ARM virtual machines provided by github.